frontend/src/mixins/shootItem.js

```diff
@@ -236,6 +236,85 @@ export const shootItem = {
     },
     maintenancePreconditionSatisfiedMessage () {
       return get(this.maintenancePreconditionSatisfiedConstraint, 'message', 'It may not be safe to trigger maintenance for this cluster')
+    },
+    caCertificateValiditiesAcceptableConstraint () {
+      const constraints = this.shootConstraints
+      return find(constraints, ['type', 'CACertificateValiditiesAcceptable'])
+    },
+    isCACertificateValiditiesAcceptable () {
+      const status = get(this.caCertificateValiditiesAcceptableConstraint, 'status', 'True')
+      return status !== 'False'
+    },
+    caCertificateValiditiesAcceptableMessage () {
+      return get(this.caCertificateValiditiesAcceptableConstraint, 'message', 'There is at least one CA certificate which expires in less than 1y. Consider schduling a Certificate Authorities Rotation for this cluster')
+    },
+    shootStatus () {
+      return get(this.shootItem, 'status', {})
+    },
+    shootStatusCredentialsRotation () {
+      return get(this.shootStatus, 'credentials.rotation', {})
+    },
+    shootStatusCredentialsRotationAggregatedPhase () {
```

Comment:
Many things are done within the for loop which makes it rather complex, let's discuss this